openpype/hosts/max/plugins/publish/extract_thumbnail.py

```diff
@@ -35,10 +41,24 @@ def process(self, instance):
             "Writing Thumbnail to"
             " '%s' to '%s'" % (filename, tmp_staging))
         review_camera = instance.data["review_camera"]
-        with viewport_camera(review_camera):
-            preview_arg = self.set_preview_arg(
-                instance, filepath, fps, frame)
-            rt.execute(preview_arg)
+        if int(get_max_version()) >= 2024:
+            with viewport_setup_updated(review_camera):
+                preview_arg = set_preview_arg(
+                    instance, filepath, frame, frame, fps)
+                rt.execute(preview_arg)
+        else:
+            visual_style_preset = instance.data.get("visualStyleMode")
+            nitrousGraphicMgr = rt.NitrousGraphicsManager
+            viewport_setting = nitrousGraphicMgr.GetActiveViewportSetting()
+            with viewport_setup(
+                    viewport_setting,
+                    visual_style_preset,
+                    review_camera):
+                viewport_setting.VisualStyleMode = rt.Name(
+                    visual_style_preset)
+                preview_arg = set_preview_arg(
+                    instance, filepath, frame, frame, fps)
+                rt.execute(preview_arg)
```

Comment:
Couldn't this also use `render_preview_animation` but with start_frame and end_frame set to the same value to render a single frame?

Reply:
yes we can